pandas/tests/plotting/test_datetimelike.py

```diff
@@ -33,6 +34,13 @@ def setup_method(self, method):
         self.period_df = [DataFrame(np.random.randn(len(x), 3), index=x,
                                     columns=['A', 'B', 'C'])
                           for x in idx]
+        mults = [1, 23]  # setup periods with multiples of freq.rule_code
```

Comment:
Instead of adding this to the setup can you keep it self-contained to the new test you are adding and simply parametrize for DataFrame and Series?

I realize you are just copying this style from the existing tests but that makes it much harder to reason about the test than it needs to be (this module is due for a cleanup)

Reply:
Ok, could you please give me a short example? As I stated in my first comment, this is my first pull request. Some more advice may save us both time. :)
So you mean I should remove lines 37 to 45 from test_datetimelike.py and instead add the same to `test_line_plot_period_mlt_series`, as in:
```
@pytest.mark.slow
def test_line_plot_period_mlt_series(self):
    freq = [str(mlt) + frq for frq, mlt in itertools.product(freq, mults)]
    idx = [period_range('12/31/1999', freq=x, periods=100) for x in freq]
    self.period_mlt_ser = [Series(np.random.randn(len(x)), x) for x in idx]
    for s in self.period_mlt_ser:
        _check_plot_works(s.plot, s.index.freq.rule_code)
```
And the same for `test_line_plot_period_mlt_frame`.

Reply:
Agreed this is due for a cleanup, but that can wait till later. For now, a compromise: These objects are only used in one test each, so I think you can do

```
def test_line_plot_period_mlt_series(self):


```
for mult in [1, 23]:
    for base in self.freq:
        idx = period_range(..., freq=freq)
        s = Series(1, idx)
        _check_plot_works(s.plot, s.index.freq.rule_code)
```

That'll require a few other changes like making `freq` a class attribute, but should ease the later refactor.

Also, can you clarify the variable names (`mlt -> multiple`, etc.)